### DISCUSSION TITLE
Mac address conflict on sonoff R2 device

### DISCUSSION THREAD

**Discussion by MrX999** [arendst/Tasmota#21697] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
I have two sonoff R2 devices that now have a conflict and won't connect to the MQTT broker at the same time. After doing a bit of digging I realised that everything works when you disable the MQTT on either device. I then noticed that they bought have the same Mac address. I have only ever used tasmotizer locally or downloaded directly from the Tasmota web downloader. Never compiled or connect to Arduino etc. recently I replaced one with the backup of the other and stopped using the original. Now that I'm using the original with new setup I noticed the issue with the Mac address. Any idea what to do now? Is one of these now for the bin?

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by MrX999** [comment]
Just to confirm I used "backup confirmation" on the original sonoff R2 and ,"Restore configuration" on the front end of the new Tasmota device also sonoff R2. I don't know what version of Tasmota I was on at the time. It was either 8.1 or 8.5. I then upgraded to 14.0

**Comment by s-hadinger** [comment]
Can you share the MAC Address from `Status 5` command on both devices?

**Reply by sfromis** [reply]
In the very rare case of two different MAC addresses, where the first 3 bytes are different, and the last 3 bytes are the same, that could indeed cause a clash, as most functions use the last 3 bytes as "id". In that case, you should be able to get around it by not relying on the default naming scheme (which is not much recommended anyway), but do your own config of topic, host name and MqttClient.

**Reply by MrX999** [reply]
That's a great idea. Just tried it and works perfect. Just to confirm, Mac address is the exact same. It was not before as they previously worked together without issues. Hope I won't have more issues. Warning to be careful when using the restore configuration as everything appears to be restored including the Mac address. thanks for your help

**Reply by sfromis** [reply]
Are you saying that **all** 6 bytes (12 digits) of the MAC are the same? That would be bad production.

That is specifically what you see as part of the `Status 5` output, in the format of `"Mac":"11:22:33:44:55:66"`

However, if you instead are looking at the 3 bytes (6 digits) in the default naming scheme, that's a major reason to either not use backup/restore, without changing the names afterwards, or to not use the default naming scheme.

**Reply by s-hadinger** [reply]
I repeat my request. Please copy the result of `status 5` command

**Reply by MrX999** [reply, marked as answer]
Hi All

Sorry for wasting your time.
So the Mac addresses are different but the setup of the MQTT has a value from the mac address, and were the same on both devices, as I had restored the configuration from one device to another.
It is now working with another value for "Client" in the MQTT setup.

Thanks for all your help. Tasmota is an unbelievable firmware I've been using for over the last 5 years, and to this date I still can't believe how stable it is and how much you can get it to do without much work from my end. 

![image](https://github.com/arendst/Tasmota/assets/60360987/f4c89121-e1c4-4d5f-8c39-ba3d428666ac)
![image](https://github.com/arendst/Tasmota/assets/60360987/3331b9b0-fcf9-4973-a9b8-cb50500aac28)
![image](https://github.com/arendst/Tasmota/assets/60360987/c9554363-7812-40f1-84aa-c47ccb69f66c)
